frontend/src/metabase/common/components/FilterPicker/NumberFilterPicker/NumberFilterPicker.tsx

```diff
@@ -67,27 +68,35 @@ export function NumberFilterPicker({
     );
   };
 
+  const handleSubmit = (event: FormEvent) => {
```

Comment:
Something is wrong here.

1. In `master` the picker looks [very differently](https://github.com/metabase/metabase/assets/6830683/d43685ff-4a43-479a-b63c-75ed07668782), here it's a list (perhaps it's a problem in the target branch)
2. existing options are being added to the list when hitting enter
3. existing options don't get selected when hitting enter (should they)?
4. the same problem as with the `StringFilterPicker` (see my other comment)


https://github.com/metabase/metabase/assets/6830683/a9731db0-1570-4646-b46d-c618a0e942db

Reply:
Removed changes to all filters but dates from the PR.

Reply:
@kamilmielnik btw I verified and in `master` these issues exist, and I believe the only thing that's different is that the popover gets closed on enter without saving the changes